### ISSUE TITLE
Clamped alpha is preventing certain blending effects

### ISSUE THREAD

**Issue by wareya** [godotengine/godot#77360]
### Godot version

4.0.3

### System information

Windows 10, amd rx 480 gpu

### Issue description

Unlike Godot 3, Godot 4 clamps alpha values before blending. I'm not sure where in the rendering pipeline this happens, or if it's even controllable, but it makes it impossible to do certain types of blending trickery that are common in 2d games. For example, in Godot 3, you can render 50% gray with 200% alpha over anything to invert its colors. In Godot 4, this is impossible; 200% alpha acts like 100% alpha.

Godot 3:

![Godot_v3 5-stable_win64_2023-05-22_13-46-30](https://github.com/godotengine/godot/assets/585488/34bd963a-45ae-4d55-aae3-4f0afd7a3f5a)

Godot 4:

![Godot_v4 0 3-stable_win64_2023-05-22_13-47-12](https://github.com/godotengine/godot/assets/585488/a11b993c-2e86-4361-a8e3-c5ad996dfafe)

### Steps to reproduce

Add a ColorRect with the following shader:

```glsl
shader_type canvas_item;

void fragment()
{
    COLOR.rgb = vec3(0.5, 0.5, 0.5);
    COLOR.a = 2.0;
}
```

### Minimal reproduction project

[ClampedAlphaRepro.zip](https://github.com/godotengine/godot/files/11534619/ClampedAlphaRepro.zip)

**Comment by wareya** [comment]
For context on where I ran into this: I'm making a voxel level editing tool, and when configuring some kinds of materials, I have to display a grid over them. In Godot 3 I was rendering that grid with this technique, but it no longer works properly.

![image](https://github.com/godotengine/godot/assets/585488/06132f01-55c1-48b7-8c53-90dcc7b01581)

**Comment by Calinou** [comment]
Can you reproduce this in 3D?

**Comment by wareya** [comment]
Inversion works in 3D:
![image](https://github.com/godotengine/godot/assets/585488/1e6b41e0-1368-442a-a998-4af6bdcc7f93)
Alpha blending happens before anything is written to the underlying buffer, and this technique is used by a fair number of old games without HDR, so at the very least, having LDR buffers wouldn't prevent this from being fixed. This is a good pointer in the right direction, though.

**Comment by clayjohn** [comment]
This was fixed last summer by https://github.com/godotengine/godot/pull/80215. The fix is included in version 4.2!

To benefit from it, you just need to enable the `rendering/viewport/hdr_2d` project setting!

_note, the colors won't be the exact same since blending happens in linear space now. But that is an expected difference between 3.x and 4.x_
![Screenshot from 2024-04-17 21-52-44](https://github.com/godotengine/godot/assets/16521339/8e09fb3e-35d5-4ca2-be7b-cfd8170a1138)
